src/psyclone/psyGen.py

```diff
@@ -4307,14 +4327,23 @@ def search_subroutine(nodelist, searchname):
             raise InternalError("Unexpected kernel AST. Could not find "
                                 "subroutine: {0}".format(name))
 
+        arg_list = []
         try:
             sub_spec = first_type_match(subroutine.content,
                                         Fortran2003.Specification_Part)
+            arg_list = [x.string for x in subroutine.content[0].items[2].items]
         except ValueError:
             pass
         else:
             self.process_declarations(new_schedule, sub_spec.content)
 
+        try:
+            new_schedule.symbol_table.specify_argument_list(arg_list)
+        except KeyError:
+            raise InternalError("Unexpected kernel AST. The argument list of"
+                                " '{0}' do not match the variable "
+                                "declarations.".format(name))
+
```

Comment:
Good error message. Please could you tweak it slightly, e.g. "The kernel argument list '{0}' does not match the .... for kernel XXXX"

Reply:
Done